engine/src/license.h

```diff
@@ -79,4 +79,53 @@ struct MCLicenseParameters
 extern MCLicenseParameters MClicenseparameters;
 extern Boolean MCenvironmentactive;
 
+static struct { const char *tag; MCLicenseClass value; } s_class_map[] =
+{
+    { "community", kMCLicenseClassCommunity },
+    { "communityplus", kMCLicenseClassCommunityPlus },
+    { "evaluation", kMCLicenseClassEvaluation },
+    { "commercial", kMCLicenseClassCommercial },
+    { "professional evaluation", kMCLicenseClassProfessionalEvaluation },
+    { "professional", kMCLicenseClassProfessional },
+    { "", kMCLicenseClassNone }
+};
+
+inline bool MCStringToLicenseClass(MCStringRef p_class, MCLicenseClass &r_class)
+{
+    uint4 t_index;
+    for(t_index = 0; t_index < sizeof(s_class_map) / sizeof(s_class_map[0]); ++t_index)
```

Comment:
Not important, but the `uint4` could just go in the for here to save being declared on a different line to initialisation. (and with the instance below too)